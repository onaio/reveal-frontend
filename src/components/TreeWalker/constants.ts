export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
