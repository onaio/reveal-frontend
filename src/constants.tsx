// strings
export const MAP = 'map';
export type MAP = typeof MAP;
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const ACTIVE_FOCUS_INVESTIGATION = 'Active Focus Investigations';
export type ACTIVE_FOCUS_INVESTIGATION = typeof ACTIVE_FOCUS_INVESTIGATION;
export const FOCUS_AREA_INFO = 'Focus Area Information';
export type FOCUS_AREA_INFO = typeof FOCUS_AREA_INFO;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;
export const MAP_ID = 'map-1';
export type MAP_ID = typeof MAP_ID;
export const GEOJSON = 'geojson';
export type GEOJSON = typeof GEOJSON;
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export type STRINGIFIED_GEOJSON = typeof STRINGIFIED_GEOJSON;
export const FOCUS_AREA_HEADER = 'Focus Area';
export type FOCUS_AREA_HEADER = typeof FOCUS_AREA_HEADER;
export const REASON_HEADER = 'Reason';
export type REASON_HEADER = typeof REASON_HEADER;
export const STATUS_HEADER = 'Status';
export type STATUS_HEADER = typeof STATUS_HEADER;
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export type CASE_NOTIF_DATE_HEADER = typeof CASE_NOTIF_DATE_HEADER;
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export type CASE_CLASSIFICATION_HEADER = typeof CASE_CLASSIFICATION_HEADER;
export const DEFINITIONS = 'Definitions';
export type DEFINITIONS = typeof DEFINITIONS;
export const IN = 'in';
export type IN = typeof IN;
export const DISTRICT = 'District';
export type DISTRICT = typeof DISTRICT;
export const CANTON = 'Canton';
export type CANTON = typeof CANTON;
export const FI_REASON = 'FI Reason';
export type FI_REASON = typeof FI_REASON;
export const FI_STATUS = 'FI Status';
export type FI_STATUS = typeof FI_STATUS;
export const ACTIVE_INVESTIGATION = 'Active Investigation';
export type ACTIVE_INVESTIGATION = typeof ACTIVE_INVESTIGATION;
export const COMPLETE = 'Complete';
export type COMPLETE = typeof COMPLETE;
export const NO = 'No';
export type NO = typeof NO;
export const RESPONSE = 'Response';
export type RESPONSE = typeof RESPONSE;
export const MEASURE = 'Measure';
export type MEASURE = typeof MEASURE;
export const OF = 'of';
export type OF = typeof OF;
export const MARK_AS_COMPLETE = 'Mark as complete';
export type MARK_AS_COMPLETE = typeof MARK_AS_COMPLETE;
export const TARGET = 'Target';
export type TARGET = typeof TARGET;
export const POLYGON = 'Polygon';
export type POLYGON = typeof POLYGON;
export const MULTI_POLYGON = 'MultiPolygon';
export type MULTI_POLYGON = typeof MULTI_POLYGON;
export const POINT = 'Point';
export type POINT = typeof POINT;
export const FEATURE = 'Feature';
export type FEATURE = typeof FEATURE;
export const FEATURE_COLLECTION = 'FeatureCollection';
export type FEATURE_COLLECTION = typeof FEATURE_COLLECTION;
export const MAIN_PLAN = 'main-plan-layer';
export type MAIN_PLAN = typeof MAIN_PLAN;
export const APP = 'APP';
export type APP = typeof APP;
export const NO_GEOMETRIES_RESPONSE = 'Goals have no Geometries';
export type NO_GEOMETRIES_RESPONSE = typeof NO_GEOMETRIES_RESPONSE;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_SINGLE_URL = '/focus-investigation/view';
export type FI_SINGLE_URL = typeof FI_SINGLE_URL;
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export type FI_SINGLE_MAP_URL = typeof FI_SINGLE_MAP_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// task action codes
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export type RACD_REGISTER_FAMILY_CODE = typeof RACD_REGISTER_FAMILY_CODE;
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export type CASE_CONFIRMATION_CODE = typeof CASE_CONFIRMATION_CODE;
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export type MOSQUITO_COLLECTION_CODE = typeof MOSQUITO_COLLECTION_CODE;
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export type BEDNET_DISTRIBUTION_CODE = typeof BEDNET_DISTRIBUTION_CODE;
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export type LARVAL_DIPPING_CODE = typeof LARVAL_DIPPING_CODE;
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export type BLOOD_SCREENING_CODE = typeof BLOOD_SCREENING_CODE;
export const IRS_CODE = 'IRS';
export type IRS = typeof IRS_CODE;
