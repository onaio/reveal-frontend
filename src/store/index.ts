import { store } from '@opensrp/store';

export default store;
