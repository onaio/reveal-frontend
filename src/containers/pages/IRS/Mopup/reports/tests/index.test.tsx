import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount, shallow } from 'enzyme';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedMopup, { Mopup } from '..';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../../constants';
import store from '../../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../../store/ducks/generic/jurisdictions';
import IRSPlansReducer, {
  genericFetchPlans,
  reducerName as IRSPlansReducerName,
} from '../../../../../../store/ducks/generic/plans';
import { mopupFocusAreas, mopupIRSJurisdictions, mopupPlans } from './fixtures/fixtures';

jest.mock('../../../../../../configs/env');

/** register the reducers */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
/** register the reducers */
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);

const mock: any = jest.fn();
const history = createBrowserHistory();

const props = {
  history,
  location: mock,
  match: {
    isExact: true,
    params: {
      planId: '7e4aba56-d5cc-5b3c-9e01-966e757c389e',
    },
    path: `${IRS_MOP_UP_REPORT_URL}/:planId`,
    url: `${IRS_MOP_UP_REPORT_URL}/7e4aba56-d5cc-5b3c-9e01-966e757c389e`,
  },
};

describe('containers/pages/IRS/Mopup/reports/', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(
      <Router history={history}>
        <Mopup {...props} />
      </Router>
    );
  });

  it('works with store', async () => {
    // populate the store
    store.dispatch(genericFetchPlans(mopupPlans));
    store.dispatch(fetchGenericJurisdictions('200', mopupIRSJurisdictions));
    store.dispatch(fetchGenericJurisdictions('201', mopupFocusAreas));

    const supersetServiceMock: any = jest.fn(async () => []);
    (props as any).service = supersetServiceMock;
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMopup {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(wrapper.find('.page-title').text()).toEqual(
      'IRS Mop-up Reporting: VL Zambia 2020 Traning Plan'
    );
    wrapper.find('.breadcrumb li').forEach((list, index) => {
      expect(list.text()).toMatchSnapshot(` ${index + 1}`);
    });
    wrapper.find('.th').forEach((header, index) => {
      expect(header.text()).toMatchSnapshot(` table headers${index + 1}`);
    });
    wrapper.find('.tr .td').forEach((data, index) => {
      expect(data.text()).toMatchSnapshot(` ${index + 1}`);
    });

    // drill down to district level.
    expect(
      wrapper
        .find('.tbody .tr')
        .at(0)
        .text()
    ).toMatchInlineSnapshot(`"Zambia(vl 2020)7629230071521"`);

    wrapper
      .find('.tbody .tr')
      .at(0)
      .find('.td')
      .at(0)
      .simulate('click', { button: 0 });

    wrapper.update();
    expect(
      wrapper
        .find('.tbody .tr')
        .at(0)
        .text()
    ).toMatchInlineSnapshot(`"Eastern Prov (2020)2844076270"`);

    wrapper
      .find('.tbody .tr')
      .at(0)
      .find('.td')
      .at(0)
      .simulate('click', { button: 0 });

    wrapper.update();
    expect(
      wrapper
        .find('.tbody .tr')
        .at(0)
        .text()
    ).toMatchInlineSnapshot(`"Chadiza District (2020)2844076270"`);

    wrapper
      .find('.tbody .tr')
      .at(0)
      .find('.td')
      .at(0)
      .simulate('click', { button: 0 });

    wrapper.update();
    expect(
      wrapper
        .find('.tbody .tr')
        .at(0)
        .text()
    ).toMatchInlineSnapshot(`"Chadiza(2020)1693522233"`);
    wrapper.find('.tr').forEach(data => {
      expect(data.text()).toMatchSnapshot(``);
    });

    wrapper
      .find('.tbody .tr')
      .at(0)
      .find('.td')
      .at(0)
      .simulate('click', { button: 0 });

    wrapper.update();
    expect(
      wrapper
        .find('.tbody .tr')
        .at(0)
        .text()
    ).toMatchInlineSnapshot(`"Chadiza(2020) (other)0000"`);

    wrapper.find('.tr').forEach(data => {
      expect(data.text()).toMatchSnapshot(`at District level`);
    });

    wrapper
      .find('.tbody .tr')
      .at(1)
      .find('.td')
      .at(0)
      .simulate('click', { button: 0 });

    wrapper.update();
    expect(
      wrapper
        .find('.tbody .tr')
        .at(1)
        .text()
    ).toMatchInlineSnapshot(
      `"CDZ_18(2020)Chadiza(2020)4103730.00%0.00%0.00%Not visitedNo decision form"`
    );

    wrapper.find('.tr').forEach(data => {
      expect(data.text()).toMatchSnapshot(`at focus Area level`);
    });
  });
});
