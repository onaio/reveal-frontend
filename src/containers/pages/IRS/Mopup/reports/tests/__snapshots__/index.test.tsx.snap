// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/IRS/Mopup/reports/ works with store 5`] = `"NameHealth Centers to Mop-upSpray Areas to to Mop-upStructures to spray or areas to react 90%TLA days needed for areas needed to reach 90%"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store 6`] = `"Chadiza(2020)1693522233"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store 7`] = `"Chadiza District (2020) (other)0000"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store 8`] = `"Sinalo(2020)11555437"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store 9`] = `"Chadiza District (2020) (other)00"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  1 1`] = `"Home"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  1 2`] = `"Zambia(vl 2020)"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  2 1`] = `"IRS Mop-up Reporting"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  2 2`] = `"7"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  3 1`] = `"VL Zambia 2020 Traning Plan"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  3 2`] = `"629"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  4 1`] = `"23007"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  5 1`] = `"1521"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  table headers1 1`] = `"Name"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  table headers2 1`] = `"Health Centers to Mop-up"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  table headers3 1`] = `"Spray Areas to to Mop-up"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  table headers4 1`] = `"Structures to spray or areas to react 90%"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store:  table headers5 1`] = `"TLA days needed for areas needed to reach 90%"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 1`] = `"NameHealth Centers to Mop-upSpray Areas to to Mop-upStructures to spray or areas to react 90%TLA days needed for areas needed to reach 90%"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 2`] = `"Chadiza(2020) (other)0000"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 3`] = `"Chadiza Central NHC11682655"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 4`] = `"Changaya NHC11433122"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 5`] = `"Kamwala1865444"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 6`] = `"Kapachi NHC11340727"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 7`] = `"Mbingwa NHC1323316"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 8`] = `"Mlolo NHC1260141"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 9`] = `"Tigwilizane NHC1921414"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 10`] = `"Yobe NHC1425617"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at District level 11`] = `"Chadiza(2020) (other)00"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 1`] = `"Health FacilitySpray AreaStructures on GroundVisited SprayedStructures remaining to spray to reach 90% SETLA DaysSpray EffectivenessFound coverageSpray coverageDate of Last VisitDate of last Decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 2`] = `"CDZ_11(2020)Chadiza(2020)9508660.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 3`] = `"CDZ_18(2020)Chadiza(2020)4103730.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 4`] = `"CDZ_31(2020)Chadiza(2020)1101010.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 5`] = `"CDZ_3(2020)Chadiza(2020)38421325225.47%8.85%5.47%21-Sep21-Sep"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 6`] = `"CDZ_33(2020)Chadiza(2020)100910.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 7`] = `"CDZ_36(2020)Chadiza(2020)90910.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 8`] = `"CDZ_45(2020)Chadiza(2020)70710.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 9`] = `"CDZ_47(2020)Chadiza(2020)70710.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 10`] = `"CDZ_49(2020)Chadiza(2020)70710.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 11`] = `"CDZ_53(2020)Chadiza(2020)60610.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 12`] = `"CDZ_59(2020)Chadiza(2020)40410.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 13`] = `"CDZ_61(2020)Chadiza(2020)40410.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 14`] = `"CDZ_63(2020)Chadiza(2020)30310.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 15`] = `"CDZ_66(2020)Chadiza(2020)20210.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 16`] = `"CDZ_7(2020)Chadiza(2020)2060186130.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 17`] = `"CDZ_9(2020)Chadiza(2020)137012490.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 18`] = `"Chadiza Central NHC (other)Chadiza(2020)00000.00%0.00%0.00%Not visitedNo decision form"`;

exports[`containers/pages/IRS/Mopup/reports/ works with store: at focus Area level 19`] = `"Chadiza Central NHC_OtherChadiza(2020)00000.00%0.00%0.00%Not visitedNo decision form"`;
